### PR TITLE
src/py/pyproject.toml: Move metadata here from setup.cfg

### PR DESCRIPTION
**Pull request by mkoeppe**
### Description

Modernizing metadata

### CODE REVIEW

File: src/py/pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+name = "pyodide-py"
+version = "0.26.0.dev0"
+authors = [{name = "Pyodide developers"}]
+description = "A Python package providing core interpreter functionality for Pyodide"
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
+    "Operating System :: OS Independent",
+]
+requires-python = ">=3.8"
```
Comment (hoodmane) [inline src/py/pyproject.toml:15]: This is quite out of date... Probably this would be closer to accurate:
```suggestion
requires-python = ">=3.11"
```

**Review by hoodmane — APPROVED** [review]
Looks good to me. I guess when we release our next alpha we'll find out if this changed anything noticeable. Probably I should do a quick before and after comparison of the wheel contents...

Thanks again @mkoeppe for helping with these chores!

**Comment by hoodmane** [comment]
Okay, well the file set is exactly the same for both the sdist and wheels. I was expecting `setup.cfg` to go away in the sdist but setuptools apparently makes one with:
```
[egg_info]
tag_build = 
tag_date = 0
```
Other than the sdist changes that we'd obviously expect, the difference between the wheels is:
```patch
 Metadata-Version: 2.1
 Name: pyodide-py
 Version: 0.26.0.dev0
-Summary: "A Python package providing core interpreter functionality for Pyodide."
-Home-page: https://github.com/pyodide/pyodide
+Summary: A Python package providing core interpreter functionality for Pyodide
 Author: Pyodide developers
+Project-URL: Homepage, https://github.com/pyodide/pyodide
 Project-URL: Bug Tracker, https://github.com/pyodide/pyodide/issues
 Project-URL: Documentation, https://pyodide.org/en/stable/
 Classifier: Programming Language :: Python :: 3
 Classifier: License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)
 Classifier: Operating System :: OS Independent
-Requires-Python: >=3.8
+Requires-Python: >=3.11
 Description-Content-Type: text/markdown
```

**Comment by mkoeppe** [comment]
Thanks for checking this carefully!

**Pull request merged**